PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -50,33 +79,130 @@ public void onResult(Configuration configuration, Exception error) {
+                    }
+                },
+                new OnError() {
+                    @Override
+                    public void onError(final ErrorInfo errorInfo) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.error");
+                        callback.onResult(null, new BraintreeException(errorInfo.getReason()));
+                    }
+                }
+        );
+    }
+
+    //TODO: I copied all of this from PayPal client and adapted it. Is all of this necessary?
+    private void approved(final Approval approval, final PayPalResponse payPalResponse, final PayPalNativeFlowCallback callback) {
+        Uri deepLinkUri = Uri.parse(String.format(
+                "%s://onetouch/v1/success?paymentId=%s&token=%s&PayerID=%s",
+                braintreeClient.getReturnUrlScheme(),
+                approval.getData().getPaymentId(),
+                approval.getData().getOrderId(),
+                approval.getData().getPayerId()));
+        try {
+            JSONObject urlResponseData = parseUrlResponseData(deepLinkUri, payPalResponse.getSuccessUrl(), payPalResponse.getApprovalUrl(), "token");
+            PayPalAccount payPalAccount = new PayPalAccount();
+            payPalAccount.setClientMetadataId(payPalResponse.getClientMetadataId());
+            payPalAccount.setSource("paypal-browser"); // paypal native throws error
+            payPalAccount.setUrlResponseData(urlResponseData);
+            payPalAccount.setPaymentType("single-payment"); // is this the correct way?
+
+            if (payPalResponse.getMerchantAccountId() != null) {
+                payPalAccount.setMerchantAccountId(payPalResponse.getMerchantAccountId());
+            }
+
+            if (payPalResponse.getIntent() != null) {
+                payPalAccount.setIntent(payPalResponse.getIntent());
+            }
+            tokenizationClient.tokenizeREST(payPalAccount, new TokenizeCallback() {
+                @Override
+                public void onResult(JSONObject tokenizationResponse, Exception exception) {
+                    if (tokenizationResponse != null) {
+                        try {
+                            PayPalAccountNonce payPalAccountNonce = PayPalAccountNonce.fromJSON(tokenizationResponse);
+                            if (payPalAccountNonce.getCreditFinancing() != null) {
+                                braintreeClient.sendAnalyticsEvent("paypal.credit.accepted");
+                            }
+                            callback.onResult(payPalAccountNonce, null);
+                        } catch (JSONException e) {
+                            callback.onResult(null, e);
+                        }
+                    } else {
+                        callback.onResult(null, exception);
+                    }
+                }
+            });
+        } catch (final JSONException | BraintreeException e) {
+            callback.onResult(null, e);
+        }
```

Comment:
I took all of this from `PayPal` module. I wonder if its completely necessary or if it even applies here.
It was the only way to go from `Approval` object that I get from the SDK, to `PaymentMethodNonce`. Perhaps there is a simpler. Do we need to pass through `tokenizationClient.tokenizeREST()`?

Reply:
I think we need to be returning a `PaymentMethodNonce` to the merchant so that they can send it to their servers to create transactions.  Using the Approval object info to create a tokenization request which returns a nonce seems like the correct flow to me. We can also look into how this is being done on iOS, as it should probably align between the two.

Reply:
I took i look at the repo and it does, they create the tokenization request with the approval information